docs/running-rasa-with-docker.rst

```diff
@@ -0,0 +1,436 @@
+    information see :ref:`cli-usage`.
+
+In this case, we've also passed values for the location of the domain file, stories file, and the models output
+directory to show how these can be customized.
+You can also leave these out since we are passing the default values.
+
+.. note::
+
+    If you are using a custom NLU component or policy, you have to add the module file to your
+    Docker container. You can do this by either mounting the file or by including it in your
+    own custom image (e.g. if the custom component or policy has extra dependencies). Make sure
+    that your module is in the Python module search path, e.g. by setting the
+    environment variable ``PYTHONPATH=$PYTHONPATH:<directory of your module>``.
+
+
+Running the Rasa Server
+-----------------------
+
+To run your AI assistant in production, configure your required
+:ref:`messaging-and-voice-channels` in ``credentials.yml``. If this file does not
+exist, create it using:
+
+.. code-block:: bash
+
+  touch credentials.yml
+
+Then edit it according to your connected channels.
+After this run the trained model with:
+
+.. code-block:: bash
+
+  docker run \
+    -v $(pwd)/models:/app/models \
+    rasa/rasa:latest-full \
+    run
+
+Command Description:
+
+  - ``-v $(pwd)/models:/app/models``: Mounts the directory with the trained Rasa model
+    in the container
+  - ``rasa/rasa:latest-full``: Use the Rasa image with the tag ``latest-full``
+  - ``run``: Executes the ``rasa run`` command. For more information see
+    :ref:`cli-usage`.
+
+
+Using Docker Compose to Run Multiple Services
+---------------------------------------------
+
+To run Rasa together with other services, such as a server for custom actions, it is
+recommend to use `Docker Compose <https://docs.docker.com/compose/>`_.
+Docker Compose provides an easy way to run multiple containers together without
+having to run multiple commands.
+
+Start by creating a file called ``docker-compose.yml``:
+
+.. code-block:: bash
+
+  touch docker-compose.yml
+
+Add the following content to the file:
+
+.. code-block:: yaml
+
+  version: '3.0'
+  services:
+    rasa:
+      image: rasa/rasa:latest-full
+      ports:
+        - 5005:5005
+      volumes:
+        - ./:/app
+      command:
+        - run
+
+
+The file starts with the version of the Docker Compose specification that you
+want to use.
+Each container is declared as a ``service`` within the docker-compose file.
+The first service is the ``rasa`` service.
+
+The command is similar to the ``docker run`` command in :ref:`running_docker_container`.
+The ``ports`` part defines a port mapping between the container and your host
+system. In this case it makes ``5005`` of the ``rasa`` service available on
+port ``5005`` of your host.
+This is the port of the :ref:`rest_channels` interface of Rasa.
+
+.. note::
+
+    Since Docker Compose starts a set of Docker containers, it is no longer
+    possible to directly connect to one single container after executing the
```

Comment:
```suggestion
    possible to interactively connect to one single container after executing the
```

Reply:
is that word @erohmensing ?

Reply:
Hmmm it's a word but not sure if it's what we want sentence-wise

Reply:
what do you mean by connect interactively? use the command line for user input? or something different

Reply:
and what do we want, Ella? :-D

Reply:
>  use the command line for user input
yes exactly that

Reply:
it's no longer possible to use the `it` flag to connect to the command line of a container

Reply:
done 👍